File: src/test/fixtures/kotlin/cognitiveComplexity.kt
fun cleanSimpleFunction(x: Int): Int {
    if (x > 0) {
        return 1
    }
    return 0
}

fun flaggedComplexFunction(x: Int): Int {
    if (x > 0) {
        if (x > 1) {
            if (x > 2) {
                if (x > 3) {
                    if (x > 4) {
                        if (x > 5) {
                            return x
                        }
                    }
                }
            }
        }
    }
    return 0
}

fun flaggedSevereFunction(x: Int): Int {
    if (x > 0) {
        if (x > 1) {
            if (x > 2) {
                if (x > 3) {
                    if (x > 4) {
                        if (x > 5) {
                            if (x > 6) {
                                return x
                            }
                        }
                    }
                }
            }
        }
    }
    return 0
}

File: src/test/fixtures/fsharp/CognitiveComplexity.fs
module CognitiveComplexity

// clean — not flagged by cognitive complexity
let cleanSimpleFunction (x: int) = if x > 0 then 1 else 0

// flagged — cognitive complexity (medium)
let flaggedComplexFunction (x: int) =
    if x > 0 then
        if x > 1 then
            if x > 2 then
                if x > 3 then
                    if x > 4 then x else 0
                else
                    0
            else
                0
        else
            0
    else
        0

// flagged — cognitive complexity (high)
let flaggedSevereFunction (x: int) =
    if x > 0 then
        if x > 1 then
            if x > 2 then
                if x > 3 then
                    if x > 4 then
                        if x > 5 then
                            if x > 6 then x else 0
                        else
                            0
                    else
                        0
                else
                    0
            else
                0
        else
            0
    else
        0
